sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClient.cs

```diff
@@ -52,22 +52,54 @@ public CryptographyClient(Uri keyId, TokenCredential credential)
         /// If you have a key <see cref="Uri"/>, use <see cref="KeyVaultKeyIdentifier"/> to parse the <see cref="KeyVaultKeyIdentifier.VaultUri"/> and other information.
         /// </param>
         /// <param name="credential">A <see cref="TokenCredential"/> used to authenticate requests to the vault, like DefaultAzureCredential.</param>
-        /// <param name="options"><see cref="CryptographyClientOptions"/> that allow to configure the management of the request sent to Key Vault.</param>
+        /// <param name="options"><see cref="CryptographyClientOptions"/> the <see cref="CryptographyClient"/> for local or remote operations on Key Vault.</param>
         /// <exception cref="ArgumentNullException"><paramref name="keyId"/> or <paramref name="credential"/> is null.</exception>
         /// <exception cref="NotSupportedException">The <see cref="CryptographyClientOptions.Version"/> is not supported.</exception>
-        public CryptographyClient(Uri keyId, TokenCredential credential, CryptographyClientOptions options) : this(keyId, credential, options, false)
+        public CryptographyClient(Uri keyId, TokenCredential credential, CryptographyClientOptions options)
+            : this(keyId, credential, options, false)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CryptographyClient"/> class.
+        /// Cryptographic operations will be performed only on the local machine.
+        /// </summary>
+        /// <param name="key">A <see cref="JsonWebKey"/> used for local cryptographic operations.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is null.</exception>
+        /// <exception cref="NotSupportedException">The <see cref="JsonWebKey.KeyType"/> of <paramref name="key"/> is not supported.</exception>
+        public CryptographyClient(JsonWebKey key)
+            : this(key, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CryptographyClient"/> class.
+        /// Cryptographic operations will be performed only on the local machine.
+        /// </summary>
+        /// <param name="key">A <see cref="JsonWebKey"/> used for local cryptographic operations.</param>
+        /// <param name="options"><see cref="LocalCryptographyClientOptions"/> to configure the <see cref="CryptographyClient"/> for local-only operations.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is null.</exception>
+        /// <exception cref="NotSupportedException">The <see cref="JsonWebKey.KeyType"/> of <paramref name="key"/> is not supported.</exception>
+        public CryptographyClient(JsonWebKey key, LocalCryptographyClientOptions options)
+        {
+            Argument.AssertNotNull(key, nameof(key));
+
+            _keyId = key.Id;
```

Comment:
Do we need to validate that `key.Id` is not null or empty?

Reply:
No. In a JWK, the [`kid` is optional](https://tools.ietf.org/html/rfc7517#section-4.5).

Reply:
BTW, I checked that everywhere we use this - locally, at least - can be null, like `DiagnosticScope.AddAttribute` (it can be null).